eureka-client/src/main/java/com/netflix/discovery/shared/resolver/EurekaEndpoint.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.discovery.shared.resolver;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Tomasz Bak
+ */
+public class EurekaEndpoint implements Comparable<EurekaEndpoint> {
```

Comment:
This Endpoint class seems pretty aws specific (references to zones etc). Can we interface this and then move the aws specific implementations of endpoint + resolver into a .aws package?